core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/StackSyncService.java

```diff
@@ -144,6 +150,22 @@ private void handleSyncResult(Stack stack, Map<InstanceSyncState, Integer> insta
         return instanceStates;
     }
 
+    private AmbariClient getAmbariClient(Stack stack) throws Exception {
+        AmbariHosts ambariHosts = ambariClusterConnector.getAmbariHosts(stack);
+        return ambariHosts.getAmbariClient();
+    }
```

Comment:
Why is it needed to use the AmbariHosts dto class instead of calling ambariClientProvider.getAmbariClient directly?